### ISSUE TITLE
Bug: Call to a member function map() on null - not yet resolved!

### ISSUE THREAD

**Issue by LeigerGaming** [spatie/nova-tags-field#112]
Issue #94 is still ongoing.

Does anyone have a workaround for this?

spatie/laravel-tags 2.7.2
spatie/nova-tags-field 3.1.1

Laravel Framework 8.40.0

**Call to a member function map() on null**
<snipped>/vendor/spatie/nova-tags-field/src/Tags.php#111
```php
            $tags = $tags->where('type', $this->meta()['type']);
        }
        return $tags->map(function (Tag $tag) {
            return e($tag->name);
        })->values();
    }
```

**Comment by LeigerGaming** [comment]
This prevents any view that it is present on from loading, as it's a PHP Exception.

I can `hideFromIndex()` and `hideFromDetail()` to work around the issue there, but the only way for me to get the update view to work is to completely uninstall the package. Does it still work, and is it still maintained as of April 2021?

**Comment by LeigerGaming** [comment]
This looks to be the full source of the function with the problem:

**vendor/spatie/nova-tags-field/src/Tags.php**
```php
    public function resolveAttribute($resource, $attribute = null)
    {
        $tags = $resource->tags;

        if (Arr::has($this->meta(), 'type')) {
            $tags = $tags->where('type', $this->meta()['type']);
        }

        return $tags->map(function (Tag $tag) {
            return $tag->name;
        })->values();
    }
```

**Comment by LeigerGaming** [comment]
This issue is now closed.

The cause of the issue was the existence of a database field called 'tags' on the related model. Nova was detecting that and it was returning null due to no value in that column, rather than using the relationship to this project's Tag model.

**The solution was a good night's sleep and a clear mind :)**

Today, I ran a `php artisan migrate:fresh` to clear my database of the previous testing field I'd created before installing spatie/laravel-tags & spatie/nova-tags-field, and that means that things now work correctly - and I've deleted my fork and closed the pull request as well.

**Note to anyone else encountering this issue:**

- Make sure you're running the latest version of all packages
- Make sure you don't have an existing column named 'tags' on the model that you're linking to :)
